src/interfaces/chain.cpp

```diff
@@ -375,9 +377,16 @@ class ChainImpl : public Chain
             notifications.TransactionAddedToMempool(entry.GetSharedTx());
         }
     }
+    std::unique_ptr<Handler> addClient(ChainClient& client) override
+    {
+        ChainClients& clients = m_node.chain_clients;
+        ChainClients::iterator it = clients.emplace(clients.end(), client);
+        return MakeHandler([&clients, it] { clients.erase(it); });
```

Comment:
Is it better to capture member `chain_clients` than `this`?

Also, it would be nice to explain that a `std::list` is used otherwise this `.erase` would fail.